src/pyhf/infer/calculators.py

```diff
@@ -699,14 +699,26 @@ def distributions(self, poi_test, track_progress=None):
         tensorlib, _ = get_backend()
         sample_shape = (self.ntoys,)
 
-        signal_pars = self.pdf.config.suggested_init()
-        signal_pars[self.pdf.config.poi_index] = poi_test
-        signal_pdf = self.pdf.make_pdf(tensorlib.astensor(signal_pars))
+        signal_pars = fixed_poi_fit(
+            poi_test,
+            self.data,
+            self.pdf,
+            self.init_pars,
+            self.par_bounds,
+            self.fixed_params,
+        )
+        signal_pdf = self.pdf.make_pdf(signal_pars)
         signal_sample = signal_pdf.sample(sample_shape)
 
-        bkg_pars = self.pdf.config.suggested_init()
-        bkg_pars[self.pdf.config.poi_index] = 1.0 if self.test_stat == 'q0' else 0.0
-        bkg_pdf = self.pdf.make_pdf(tensorlib.astensor(bkg_pars))
+        bkg_pars = fixed_poi_fit(
+            1.0 if self.test_stat == 'q0' else 0.0,
+            self.data,
+            self.pdf,
+            self.init_pars,
+            self.par_bounds,
+            self.fixed_params,
+        )
+        bkg_pdf = self.pdf.make_pdf(bkg_pars)
         bkg_sample = bkg_pdf.sample(sample_shape)
```

Comment:
what is the ROOT imlplementation here? are two fits performed or is the data fitted and those NP values are used for both signal and background?

Reply:
( @masonproffitt for the notification)

Reply:
I'm far from a RooStats expert, but it looks like the default behavior is to separately fit the NPs both for the [null hypothesis](https://root.cern.ch/doc/master/FrequentistCalculator_8cxx_source.html#l00105) and for the [alternative hypothesis](https://root.cern.ch/doc/master/FrequentistCalculator_8cxx_source.html#l00218). This is also the ATLAS recommendation (in the document linked from the original issue).

Reply:
thanks @masonproffitt  - can you add a link to this doccument / explanation to the docstring?

Reply:
Okay, i've added a note about this and a link